presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -174,6 +180,22 @@ private Expression simplifyExpression(Expression expression)
             ExpressionInterpreter interpreter = ExpressionInterpreter.expressionOptimizer(expression, metadata, session, expressionTypes);
             return LiteralInterpreter.toExpression(interpreter.optimize(NoOpSymbolResolver.INSTANCE), expressionTypes.get(NodeRef.of(expression)));
         }
+
+        private Expression simplifyFilterPredicate(Expression predicate)
```

Comment:
simplify is a wrong name as it does not tell anything. I would name it `replaceSelfEqualityWithNotNull`.